### DISCUSSION TITLE
Maximum Message Size

### DISCUSSION THREAD

**Discussion by ussefdesouky** [eProsima/Fast-DDS#3813] [Q&A]
Hello Community,
I want to know the message size limitations for all the transport layers(UDP, TCP, SHM & ZeroCopy). From the documentation I have seen that the maximum message size is 64Kb, but I have published an array of size [20446272] and it worked with the mentioned protocols, so I need to know what are the constraints or the points of failure when sending this size of arrays.

One more questions [https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/large_data/large_data.html](url) for the _15.3.6. Practical Examples_ it's mentioned that he is sending a file of 9mb and the message size is 64Kb so he needs 1100 fragment, but per my understanding the fragements = 9mb / 64kb = 156 fragments. That the message is divided into 156 fragment.

Correct me If I am wrong.
Thanks.

**Comment by JLBuenoLopez** [comment]
@ussefdesouky 

You may find of interest [this comment](https://github.com/eProsima/Fast-DDS/issues/3053#issuecomment-1303094535). You need to differentiate between the different transport layers and protocols being used. 64kb is the UDP datagram maximum size. Fast DDS fragmentation protocol would kick in if your message size is bigger than that. On the other hand, IP protocol would fragment the UDP datagram in order to fit it within the network MTU. Consequently, there is no limit on the message size, though the bigger the message the more probable that a fragment (either from Fast DDS or IP protocol) is lost and the message is not received. But that depends completely on the network architecture and its reliability.

**Reply by ussefdesouky** [reply]
So from the above mentioned protocols can you provide me with the most reliable and efficient for sending large data.
